internal/ceb/app_config.go

```diff
@@ -79,33 +159,50 @@ func (ceb *CEB) watchAppConfig(
 			// New variables, track it and immediately trigger a refresh
 			log.Debug("new config variables received, scheduling refresh")
 			prevVars = newVars
-			refreshChNow := make(chan time.Time)
-			close(refreshChNow)
-			refreshCh = refreshChNow
+			refreshTick()
 
 			// Split the static and dynamic out here since this is something
 			// we're going to need often so we precompute it once.
 			dynamicOld := dynamic
 			static, dynamic = ceb.splitAppConfig(log, newVars)
 
 			// We need to do a diff of if any dynamic var config changed.
-			prevVarsChanged = ceb.diffDynamicAppConfig(log, dynamicOld, dynamic)
+			// We loop through the result here and set values to true so
+			// that we don't clobber changes that inSourceCh receiving may have
+			// set. On refresh, we always reset prevVarsChanged to empty.
+			for k, v := range ceb.diffDynamicAppConfig(log, dynamicOld, dynamic) {
+				// If it is false, we override it with whatever v we have.
+				if !prevVarsChanged[k] {
+					prevVarsChanged[k] = v
+				}
+			}
+
+		// Case: timer fires after a period of time where we have received
+		// no other messages and we can now force a refresh.
+		case <-coalesceCh:
+			coalesceCh = nil
+			refreshNowCh := make(chan time.Time)
```

Comment:
Could we make this a global instead of a new one each time? We just want to assign a closed channel, right? Mostly trying to make the logic easier to understand.

Reply:
Good idea.